apollo-compiler/src/main/kotlin/com/apollostack/compiler/ir/InlineFragment.kt

```diff
@@ -0,0 +1,24 @@
+package com.apollostack.compiler.ir
+
+import com.squareup.javapoet.TypeSpec
+import javax.lang.model.element.Modifier
+
+data class InlineFragment(
+    val typeCondition: String,
+    val fields: List<Field>
+) : CodeGenerator {
+
+  override fun toTypeSpec(): TypeSpec =
+      TypeSpec.interfaceBuilder(interfaceName())
+          .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+          .addMethods(fields.map(Field::toMethodSpec))
+          .addTypes(fields.filter(Field::isNonScalar).map(Field::toTypeSpec))
+          .build()
+
+  fun interfaceName() = "$INTERFACE_PREFIX${typeCondition.capitalize()}"
```

Comment:
Currently, all inline fragments in the IR have type conditions.

But inline fragments in GraphQL don't need to have type conditions, they can also be used to apply directives. Like in this example from the spec:

```graphql
query inlineFragmentNoType($expandedInfo: Boolean) {
  user(handle: "zuck") {
    id
    name
    ... @include(if: $expandedInfo) {
      firstName
      lastName
      birthday
    }
  }
}
```